dotnet/src/SemanticKernel.Skills/Skills.Memory.Sqlite/Database.cs

```diff
@@ -32,9 +32,10 @@ public static async Task<SqliteConnection> CreateConnectionAsync(string filename
     {
         await CreateTableAsync(conn, cancel);
 
+        //Could potentially use REPLACE instead of IGNORE here
         SqliteCommand cmd = conn.CreateCommand();
         cmd.CommandText = $@"
-             INSERT INTO {TableName}(collection, key, value, timestamp)
+             INSERT or IGNORE into {TableName}(collection, key, value, timestamp)
```

Comment:
this would turn the Insert method into an Upsert. I think it's better adding an Upsert method to the class instead

Reply:
Added both InsertAndIgnore and InsertAndReplace as separate functions

Reply:
what's the rationale for ignoring conflicts by default? shouldn't it be a developer choice? 
there are 3 scenarios:
1. on conflict update => Upsert
2. on conflict throw an error
3. on conflict ignore error

Looks like you have a preference, but other developers might need something differently. Should't we allow the flexibility?

Reply:
The flexibility exists via these functions: InsertAsync(...), InsertOrIgnoreAsync(...),  UpsertAsync(...)
which perform INSERT, INSERT or IGNORE, and INSERT or UPDATE respectively. They throw an error (sqlite throws the error not us), ignore the conflict, and update respectively.
I was defaulting to InsertOrIgnoreAsync(...) before but that has now been changed to UpsertAsync(...);